lib/logflare/backends/adaptor/bigquery_adaptor/google_api_client.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Logflare.Backends.Adaptor.BigQueryAdaptor.GoogleApiClient do
+  @moduledoc false
+  alias Explorer.DataFrame
+  alias Google.Cloud.Bigquery.Storage.V1.BigQueryWrite
+  alias Google.Cloud.Bigquery.Storage.V1.AppendRowsRequest
+  alias Google.Cloud.Bigquery.Storage.V1.AppendRowsRequest.ArrowData
+  alias Google.Cloud.Bigquery.Storage.V1.ArrowRecordBatch
+  require Logger
+
+  @finch_instance_name Logflare.FinchBQStorage
+
+  def append_rows({:arrow, data_frame}, project, dataset, table) do
+    partition_count = System.schedulers_online()
+    partition = :erlang.phash2(self(), partition_count)
+
+    {:ok, goth_token} = Goth.fetch({Logflare.Goth, partition})
+
+    {:ok, channel} =
+      GRPC.Stub.connect("https://bigquerystorage.googleapis.com",
+        adapter: GRPC.Client.Adapters.Finch,
+        adapter_opts: [instance_name: @finch_instance_name],
+        headers: [
+          {"Authorization", "Bearer #{goth_token.token}"}
+        ]
+      )
+
+    {:ok, arrow_schema} = DataFrame.dump_ipc_schema(data_frame)
+
+    writer_schema = %Google.Cloud.Bigquery.Storage.V1.ArrowSchema{serialized_schema: arrow_schema}
+
+    {:ok, batch_msgs} = DataFrame.dump_ipc_record_batch(data_frame)
```

Comment:
What is the chance that it would actually be split up into many batch messages? Given that Broadway will already batch it to a set number of events.
Using the Task to fire of the requests might be heavier than just firing it within the process due to copying needed

Reply:
In event that it does split it into multiple RecordBatch then we can emit a warning (since it shouldn't be happening) and can remove the Task overhead by doing both sequentially. Don't need to factor it in since it is an edge case.